services/terraformer/server/loadbalancer/dns.go

```diff
@@ -162,14 +163,17 @@ func (d DNS) generateFiles(dnsID, dnsDir string, dns *pb.DNS, nodeIPs []string)
 		return fmt.Errorf("error creating provider credential key file for provider %s in %s : %w", dns.Provider.SpecName, dnsDir, err)
 	}
 
-	templateLoader := templateUtils.TemplateLoader{Directory: templateUtils.TerraformerTemplates}
-	tpl, err := templateLoader.LoadTemplate(fmt.Sprintf("%s-dns.tpl", dns.Provider.CloudProviderName))
+	sourceDirectory := templateUtils.TemplateLoader{
+		Directory: path.Join(templateUtils.TerraformerTemplates, dns.GetProvider().GetCloudProviderName(), "dns"),
```

Comment:
I think this line will break if the provider is `hetznerdns`, as it will form `<templates>/hetznerdns/dns` instead of `<templates>/hetzner/dns`.

Reply:
Yes, good catch!